repo_commit.go

```diff
@@ -96,8 +96,8 @@ func (repo *Repository) getCommitIdOfPackedRef(refpath string) ([]byte, error) {
 	scan := bufio.NewScanner(f)
 
 	for scan.Scan() {
-		if strings.Contains(scan.Text(), refpath) {
-			return scan.Bytes(), nil
+		if refLine, err := parseRefLine(scan.Text()); err == nil && refLine.refpath == refpath {
```

Comment:
I didn't want to assume comprehensive coverage of all possible lines in the packed ref file. This just looks for lines that match the format it recognizes (`${commit-id} ${ref-name}`) and ignores all the rest.